File: java/Factory/Maze.java
/**
 * Created by georgef on 5/29/16.
 */
import java.util.ArrayList;
import java.util.List;

public class Maze {
    private List<Room> roomList = new ArrayList<>();
    private List<Wall> wallList = new ArrayList<>();

    public void addRoom(Room room) {
        roomList.add(room);
    }
    public void addWall(Wall wall) {
        wallList.add(wall);
    }
}

File: csharp/Twin_Simplified/Maze.cs
using System.Collections.Generic;
using System.Linq;

namespace DesignPatterns.Twin_Simplified
{
    class Maze
    {
        List<Room> rooms = new List<Room>();
        List<Wall> walls = new List<Wall>();

        internal void addRoom(Room room)
        {
            rooms.Add(room);
        }

        internal void addWall(Wall wall)
        {
            walls.Add(wall);
        }
    }

}
